nixos/modules/programs/x2goserver.nix

```diff
@@ -0,0 +1,54 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.x2goserver;
+in {
+  options.programs.x2goserver.enable = mkEnableOption "x2goserver";
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ pkgs.x2goserver ];
+    users.extraGroups.x2gouser = {};
+    users.extraUsers.x2gouser = {
+      createHome = true;
+      home = "/var/lib/x2go";
+      group = "x2gouser";
+    };
+    users.extraGroups.x2goprint = {};
+
+    security.wrappers.x2gosqliteWrapper = {
+      source = "${pkgs.x2goserver}/lib/x2go/libx2go-server-db-sqlite3-wrapper";
+      owner = "root";
+      group = "x2gouser";
+      setgid = true;
+    };
+    security.wrappers.x2goprintWrapper = {
+      source = "${pkgs.x2goserver}/bin/x2goprint";
+      owner = "root";
+      group = "x2goprint";
+      setgid = true;
+    };
+    # x2goclient sends SSH commands with preset PATH set to
+    # "/usr/local/bin;/usr/bin;/bin". Since we cannot filter arbitrary ssh
+    # commands, we have to make the following executables available.
```

Comment:
Ugh, is it too difficult to patch that $PATH? I'm not thrilled about the module adding symlinks in /usr/local/bin/.

Reply:
Sadly I think this cannot be patched on the server side (unless there is a generic way to e.g. apply regex to SSH commands, but that's probably even uglier), and the PATH is hardcoded in the client in all distributions and versions.
Also, this is probably more fragile than linking the whole /run/current-system/sw/bin but I'm not 100% sure I got all used executables.

Reply:
Ok.

Reply:
As you mention `x2goclient` hard codes `PATH`. I would have thought the better route to go here would be to simply pickup `PATH` from the machine you connect to. I would suggest creating an issue upstream asking why they do not do as such. If upstream can be convinced to let the `x2goclient` client depend on the client for `PATH` (or at least append/prepend to `PATH` instead of clobbering it) then you can set `users.users.x2go.extraPackages = with pkgs; [ x2goserver xorg.xrandr xorg.xmodmap utillinux gawk coreutils ];

Reply:
I have opened a bug-report with upstream, it is awaiting moderation. Hopefully in the (probably not-so-near) future we will be able to get rid of this.

Reply:
*If* your upstream report doesn't gain acceptance is there anything stopping you from using [buildFHSUserEnv](https://nixos.org/nixpkgs/manual/#sec-fhs-environments) to set a custom shell for the `x2go` user instead of `activationScripts`?

Reply:
Unfortunately this isn't for the x2go user, the client logs in as the actual user, and maybe that's why they set the PATH so restrictive. Unless I missed something the x2go user is only to own stuff, like the session database.
Also even with cooperative upstream, the client fix wouldn't be available on other distributions until way in the future. But it's apparently still awaiting moderation on the bugs list.

Reply:
Because you didn't link to it, here is the bug report: https://bugs.x2go.org/cgi-bin/bugreport.cgi?bug=1398

I think you should give `buildFHSUserEnv` a try to get around this restriction for sure. I personally don't want to merge this PR if it screws with `/usr/bin`.

Reply:
Ah thank you for the bug link, I didn't see it go through.
As already mentioned buildFHSUserEnv does not work, because this is not about the `x2go` user, but the actual user account logging in, via standard SSH.
Also the outcome of that bug report should have no effect on this PR, because it's a client-side problem, meaning even if they fix it right now, it would only work on most distributions far in the future, and nobody on Ubuntu 18.04 or stuff like that would ever be able to use it.